src/app/services/connectivity/connectivity.service.ts

```diff
@@ -1,30 +1,51 @@
 import { Injectable } from '@angular/core';
-import { BehaviorSubject, Observable } from 'rxjs';
+
+import { ConnectableObservable, Observable } from 'rxjs';
+import { multicast, publishReplay } from 'rxjs/operators';
+
 
 @Injectable({
   providedIn: 'root'
 })
 export class ConnectivityService {
-  private monitor: Observable<boolean>;
-  private subject: BehaviorSubject<boolean>;
-
-  constructor() {
-    this.monitor = new Observable((connectivityObserver) => {
-      // map 'offline' and 'online' window events to boolean 'connnectivity state'
-      window.addEventListener('offline', () => {
-        connectivityObserver.next(false);
-      });
-      window.addEventListener('online', () => {
-        connectivityObserver.next(true);
-      });
-    });
-    // obtain initial connectivity state from navigator.onLine property
-    this.subject = new BehaviorSubject(navigator.onLine);
-    this.monitor.subscribe(this.subject);
+  public static readonly OFFLINE_EVENT = 'offline';
+  public static readonly ONLINE_EVENT = 'online';
+  private connectivityMonitor: Observable<boolean>;
+  private connectivityObserver: ConnectableObservable<boolean>;
```

Comment:
I'm big on finnish notation `connectivityMonitor$`

Reply:
Never been a fan of notation tricks.

However the variable name `ConnectivityMonitor` probably isn't all that helpful in this case and probably should be changed. `connectedStatus` is probably clearer (and preferable to `isConnected` which would falsely imply it's a primitive boolean property).